### ISSUE TITLE
No TSC error for duplicate dynamic keys with static readonly values

### ISSUE THREAD

**Issue by tompere** [Microsoft/TypeScript#26318]
**TypeScript Version:**  3.1.0-dev.20180809

**Search Terms:** dynamic key duplicate identifier

**Code**

```ts
const X = 'x'

enum Y {
  ONE = 'one',
  TWO = 'one'
}

// static key - 'duplicate identifier' error

const obj1 = {
  'y': 'wham',
  'y': 'bam' // as expected, tsc error `Duplicate identifier ''y''`
}

const obj2 = {
  y: 'wham',
  y: 'bam' // as expected, tsc error `Duplicate identifier 'y'`
}

// [the bug] dynamic key - no 'duplicate identifier' error

const obj3 = {
  [X]: 'wham',
  [X]: 'bam' // no tsc error!
}

const obj4 = {
  [Y.ONE]: 'wham',
  [Y.TWO]: 'bam' // no tsc error!
}
```

**Expected behavior:**
When assigning **duplicate dynamic keys** to plain objects, which can be statically inferred (e.g. `enum`s or `const` with primitive type) - tsc should show a compilation error.

**Actual behavior:**
no compilation error.

**Playground Link:** 
[a long long long playground link ...](https://www.typescriptlang.org/play/#src=const%20X%20%3D%20'y'%0D%0A%0D%0Aenum%20Y%20%7B%0D%0A%20%20%20%20ONE%20%3D%20'one'%2C%0D%0A%20%20%20%20TWO%20%3D%20'two'%0D%0A%7D%0D%0A%0D%0A%2F%2F%20static%20key%20-%20'duplicate%20identifier'%20error%0D%0A%0D%0Aconst%20obj1%20%3D%20%7B%0D%0A%20%20%20%20'y'%3A%20'wham'%2C%0D%0A%20%20%20%20'y'%3A%20'bam'%20%2F%2F%20as%20expected%2C%20tsc%20error%20%60Duplicate%20identifier%20''y''%60%0D%0A%7D%0D%0A%0D%0Aconst%20obj2%20%3D%20%7B%0D%0A%20%20%20%20y%3A%20'wham'%2C%0D%0A%20%20%20%20y%3A%20'bam'%20%2F%2F%20as%20expected%2C%20tsc%20error%20%60Duplicate%20identifier%20'y'%60%0D%0A%7D%0D%0A%0D%0A%2F%2F%20%5Bthe%20bug%5D%20dynamic%20key%20-%20no%20'duplicate%20identifier'%20error%0D%0A%0D%0Aconst%20obj3%20%3D%20%7B%0D%0A%20%20%20%20%5BX%5D%3A%20'wham'%2C%0D%0A%20%20%20%20%5BX%5D%3A%20'bam'%20%2F%2F%20no%20tsc%20error!%0D%0A%7D%0D%0A%0D%0Aconst%20obj4%20%3D%20%7B%0D%0A%20%20%20%20%5BY.ONE%5D%3A%20'wham'%2C%0D%0A%20%20%20%20%5BY.TWO%5D%3A%20'bam'%20%2F%2F%20no%20tsc%20error!%0D%0A%7D)

**Related Issues:**  N/A

**Comment by AlCalzone** [comment]
Why should this be an error?
```ts
enum Y {
    ONE = 'one',
    TWO = 'two'
}

const obj4 = {
    [Y.ONE]: 'wham',
    [Y.TWO]: 'bam' // no tsc error!
}
```

**Comment by tompere** [comment]
@AlCalzone you're right.
updated:
```
enum Y {
    ONE = 'one',
    TWO = 'one'
}
```

**Comment by ajafff** [comment]
Duplicate of #25758
